base/file.jl

```diff
@@ -47,7 +47,19 @@ function mkpath(path::AbstractString, mode::Unsigned=0o777)
     dir = dirname(path)
     (path == dir || isdir(path)) && return
     mkpath(dir, mode)
-    mkdir(path, mode)
+    try
+        mkdir(path, mode)
+    # If there is a problem with making the directory, but the directory
+    # does in fact exist, then ignore the error. Else re-throw it.
+    catch err
+        # isdir comes first because I figure it will return false more often
+        # than isa
```

Comment:
`isdir` is a fairly expensive syscall while `isa` is essentially a free pointer comparison. (although this is the failure case just after trying an even more expensive `mkdir` syscall, so the relative performance probably isn't too important)